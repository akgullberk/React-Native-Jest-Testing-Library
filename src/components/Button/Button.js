import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import React from 'react'

const Button = () => {
  return (
    <TouchableOpacity>
        <Text>asd</Text>
    </TouchableOpacity>
  )
}

export default Button

const styles = StyleSheet.create({})