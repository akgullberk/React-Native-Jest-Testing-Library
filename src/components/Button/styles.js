import { StyleSheet } from "react-native";

export default {
    primary: StyleSheet.create({
        container:{
            backgroundColor:"blue",
            padding:10,
            borderRadius:10,
            margin:10,
            justifyContent:"center",
            alignItems:"center",
          },
          title:{
            color:"white",
            fontWeight:"bold",
          },
}),

secondary: StyleSheet.create({
    container:{
        borderColor:"blue",
        borderWidth:1,
        padding:10,
        borderRadius:10,
        margin:10,
        justifyContent:"center",
        alignItems:"center",
      },
      title:{
        color:"blue",
        fontWeight:"bold",
      },
}),
}