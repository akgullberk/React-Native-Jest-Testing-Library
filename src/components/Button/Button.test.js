import {render} from '@testing-library/react-native'
import React from 'react'
import Button from './Button'

test('should match with snapshot', () => {
    const comp = render(<Button/>)
    expect(comp).toMatchSnapshot();
})

test('should render title correctly', () => {
    const testTitle = "test"
    const comp = render(<Button title={testTitle}/>)

    const buttonText = comp.getByTestId('button-title').children[0];
    expect(buttonText).toBe(testTitle)

    
    
    })