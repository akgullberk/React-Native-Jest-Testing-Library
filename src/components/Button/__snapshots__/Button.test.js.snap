// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text>
    asd
  </Text>
</View>
`;
